src/global_thread_handle.cu

```diff
@@ -462,6 +462,10 @@ struct MemoryManager
         , m_alloc_scaling_threshold(16 * 1024 * 1024)
     {
         //initializeCriticalSection(&m_mutex);
+
+        cudaDeviceProp devProp;
+        cudaGetDeviceProperties(&devProp, 0);
+        this->smCount = devProp.multiProcessorCount;
```

Comment:
Minor thing - Maybe we can store whole device properties structure - can be useful in some other parts of the code in the future, and calling cudaGetDeviceProperties() is slow.

Reply:
Yes, definitely - I think that will be useful too.